pydocmd/preprocessors/smart.py

```diff
@@ -0,0 +1,38 @@
+from pydocmd.preprocessors.rst import Preprocessor as RSTPreprocessor
+from pydocmd.preprocessors.google import Preprocessor as GooglePreprocessor
+
+
+class Preprocessor(object):
+    """
+    This class implements the preprocessor for restructured text and google.
```

Comment:
I like the idea of a smart preprocessor!

Reply:
Works for my project. I did not add simple preprocessor as there is a call of `link_lookup` method that is missing. I could not find it, so I could not test simple preprocessor

Reply:
Reverted pylint auto formatting, now it should be easier to see what has changed

Reply:
Thanks! The link_lookup got introduced in another PR and is a bit ugly at the moment. I believe you would just need to propagate the attribute from the smart processor object to the actual processor that will be used